crates/common/tedge_config_macros/examples/renaming_keys.rs

```diff
@@ -10,6 +10,26 @@ pub enum ReadError {
     GenericError(String),
 }
 
+pub trait AppendRemoveItem {
+    type Item;
+
+    fn append(current_value: &mut Option<Self::Item>, new_value: Self::Item);
+
+    fn remove(current_value: &mut Option<Self::Item>, remove_value: Self::Item);
+}
+
+impl<T: std::any::Any> AppendRemoveItem for T {
+    type Item = T;
+
+    fn append(_current_value: &mut Option<Self::Item>, _new_value: Self::Item) {
+        unimplemented!()
+    }
+
+    fn remove(_current_value: &mut Option<Self::Item>, _remove_value: Self::Item) {
+        unimplemented!()
+    }
+}
```

Comment:
I don't get the point of this local unimplemented trait?

Reply:
Macro calls either `append` or `remove` from `AppendRemoveItem` trait like [here](https://github.com/thin-edge/thin-edge.io/pull/2943/commits/c8f74617fd55bfdf882cfd1ae2c0d150d736f217#diff-cd95e9aea2dc9e842e4cd738eaf06c1e284ccab0754a684defe0851b188a7faeR373). We can't implement it inside the macro as it does not know about the types that are provided. Instead, local implementation of that trait is needed but we have to do that in every module that has `tedge_config` defined as sharing it between different modules is not possible. And we don't actually need to implement `append` or `remove` in those test modules because It will never be called there (but macro needs to be aware of them) thus we used `unimplemented!()`.